### DISCUSSION TITLE
Erläuterung der Bedeutung der Datenpunkte irgendwo?

### DISCUSSION THREAD

**Discussion by wangnick** [open3e/open3e#121] [Q&A] [answered]
Liebe Leute,

gibt es irgendwo eine etwas detailliertere Erläuterung der Datenpunkte, in meinem Fall für eine Vitocal 250-A?

1. In der unter https://github.com/open3e/open3e/wiki/050-Viessmann-Dokumente verlinkten Serviceanleitung sind schreibbare Parameter beschrieben, nicht aber die nur lesbaren Sensorwerte?
2. Wie kann ich mit open3e den aktuellen Wert des Parameters 1085.1 (Ausschalthysterese Speichertemperatur-Sollwert) abfragen? Wenn ich -r 1085 angebe, erhalte ich nur den Wert für 1085.0?

LG, Sebastian

**Comment by MyHomeMyData** [comment, marked as answer]
Hallo Sebastian, den besten Überblick bekommst Du mit einem Blick in die [Datenpunktliste](https://github.com/open3e/open3e/blob/develop/src/open3e/Open3Edatapoints.py) im Branch develop. Die Bezeichnungen der Datenpunkte sind meist recht sprechend.

Die Definition von 1085 ist leider im Moment nicht sinnvoll. Es ist eine Datenlänge von 4 angeben, per Codec "O3EInt16" werden aber nur 2 Bytes dekodiert.
Die Definition müsste vermutlich so aussehen:
```
1085 : O3EComplexType(4, "DomesticHotWaterHysteresis", [O3EInt6(2, "SetpointSwitchOn", signed=True), O3EInt16(2, "SetpointSwitchOff", signed=True)])
```

Wenn Du das in Deiner lokalen Datei `Open3Edatapoints.py` änderst, müsste es funktionieren. Würdest Du das mal ausprobieren? Wenn es passt, übernehme ich es in der Branch develop. Wo die open3e-Datei liegen ist [hier](https://github.com/open3e/open3e/wiki/030-Installation-und-Inbetriebnahme-von-open3E) beschrieben.

Gruß, Jürgen

**Comment by wangnick** [comment]
Hallo Jürgen, ich habe jetzt 1085 und auch gleich 1087 wie folgt geändert:
```
        1085 : O3EComplexType(4, "DomesticHotWaterHysteresis", [O3EInt16(2, "SetpointSwitchOn", signed=True), O3EInt16(2, "SetpointSwitchOff", signed=True)]),
        1087 : O3EComplexType(2, "DomesticHotWaterDuration", [O3EInt8(1, "SetpointMaxOn", signed=False), O3EInt8(1, "SetpointMinOff", signed=False)]),
```
Dann erhalte ich:
```
(.venv) vito@vitopi:~/vitocal/.venv/lib/python3.11/site-packages/open3e $ open3e -r 1085,1087 -v -j
None 1085 DomesticHotWaterHysteresis {"SetpointSwitchOn": 5.0, "SetpointSwitchOff": 0.0}
None 1087 DomesticHotWaterDuration {"SetpointMaxOn": 240.0, "SetpointMinOff": 60.0}
closing 0x680 - bye!
```

Das Zeitintervall `SetpointMaxOn` musste ich vorzeichenlos machen, sonst erhalte ich -16.0. Ich denke aber, dass wohl auch der SetpointMinOff und die Hysterese-Werte eigentlich vorzeichenlos sein sollten.

Die Einheiten der Werte sind hier wohl K und Minuten.

LG, Sebastian

**Reply by MyHomeMyData** [reply]
Das sieht doch gut aus. Dann übernehme ich das so. Bei 1087 würde ich aber bei der Bezeichnung `MaximumDomesticHotWaterLoadingTime` bleiben. Das könnte sonst Probleme machen, falls das schon jemand nutzt.

Ich baue auch gleich noch den Hinweis von [FrankSteiner](https://community.viessmann.de/t5/user/viewprofilepage/user-id/61077) aus dem Viessmann-Forum zu 497 ein.

**Reply by MyHomeMyData** [reply]
Ist im Branch develop integriert. Werde in den nächsten Tagen mal wieder einen Merge develop -> master machen.

Schönes WE!
Jürgen
